pkg/leaderelection/fake/leader_election.go

```diff
@@ -30,6 +32,20 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/recorder"
 )
 
+// ControllableResourceLockInterface is an interface that extends resourcelock.Interface to be
+// controllable.
+type ControllableResourceLockInterface interface {
+	resourcelock.Interface
+
+	// BlockLeaderElection blocks the leader election process when called. It will not be unblocked
+	// until UnblockLeaderElection is called.
+	// Not thread safe.
```

Comment:
nit: I'm not sure I follow what makes this not thread safe (same in l.46)

Or maybe the comment is outdated?